Dockerfile.riscv

```diff
@@ -0,0 +1,75 @@
+# Dockerfile.riscv
+FROM ubuntu:22.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Update and install initial prerequisites
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    build-essential \
+    git \
+    m4 \
+    scons \
+    zlib1g \
+    zlib1g-dev \
+    libprotobuf-dev \
+    protobuf-compiler \
+    libprotoc-dev \
+    libgoogle-perftools-dev \
+    python3-dev \
+    libboost-all-dev \
+    pkg-config \
+    python3-tk \
+    autoconf \
+    automake \
+    autotools-dev \
+    curl \
+    python3-pip \
+    libmpc-dev \
+    libmpfr-dev \
+    libgmp-dev \
+    gawk \
+    bison \
+    flex \
+    texinfo \
+    gperf \
+    libtool \
+    patchutils \
+    bc \
+    libexpat-dev \
+    ninja-build \
+    cmake \
+    libglib2.0-dev \
+    libslirp-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+# Clone the RISC-V GNU Toolchain
+RUN git clone --recursive https://github.com/riscv-collab/riscv-gnu-toolchain /opt/riscv-gnu-toolchain
+
+# Clone gem5
+RUN git clone --recursive https://github.com/gem5/gem5 /opt/gem5
```

Comment:
be careful. The simulation environment works for the gem5 v22.1.0.0. You are missing the checkout command